### PR TITLE
🔨(project) use docker healthcheck for database services

### PR DESCRIPTION
**Pull request by wilbrdt**
## Purpose

Docker compose provides `healthcheck` since format version 3.4.
https://docs.docker.com/compose/compose-file/compose-file-v3/#healthcheck

## Proposal

Changing from dockerize to docker healthchecks for all database services locally.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -235,9 +235,8 @@ run-all: \
 .PHONY: run-all
 
 run-clickhouse: ## start clickhouse backend
-	@$(COMPOSE) up -d clickhouse
 	@echo "Waiting for clickhouse to be up and running..."
-	@$(COMPOSE_RUN) dockerize -wait tcp://clickhouse:9000 -timeout 60s
+	@$(COMPOSE) --verbose up -d --wait clickhouse
```
Comment (SergioSim) [inline Makefile:239]: Does the `--verbose` option add logs regarding the health checks?

File: docker-compose.yml [hunk 2]
```diff
@@ -66,6 +78,10 @@ services:
       - "49178:35357"
     environment:
       KS_SWIFT_PUBLIC_URL: http://127.0.0.1:49177
+    healthcheck:
+        test: curl -s http://127.0.0.1:35357 || exit 1
+        interval: 1s
+        retries: 60
 
   # -- tools
   dockerize:
```
Comment (SergioSim) [inline docker-compose.yml:87]: Do we still need `dockerize` here?

**Review by SergioSim — APPROVED** [review]
LGTM) :+1: 
Have only minor questions/suggestions)

image

File: Makefile (hunk 1 above)
Comment (wilbrdt) [reply Makefile:239]: Nop, I forgot to remove it :grin:

File: docker-compose.yml [hunk 3]
```diff
@@ -37,11 +37,19 @@ services:
...
+      test: curl -s http://localhost:9200 || exit 1
+      interval: 1s
+      retries: 60
 
   mongo:
     image: mongo:5.0.9
     ports:
       - "27017:27017"
+    healthcheck:
+      test: echo 'db.runCommand("ping").ok' | mongosh localhost:27017/test --quiet
```
[the remark this answers is not in the document]
Comment (wilbrdt) [reply docker-compose.yml:50]: Thanks, changed!

File: docker-compose.yml [hunk 4]
```diff
@@ -37,11 +37,19 @@ services:
       memlock:
         soft: -1
         hard: -1
+    healthcheck:
+      test: curl -s http://localhost:9200 || exit 1
```
[the remark this answers is not in the document]
Comment (wilbrdt) [reply docker-compose.yml:41]: Two good remarks, thanks!

File: docker-compose.yml (hunk 2 above)
Comment (wilbrdt) [reply docker-compose.yml]: Not needed anymore indeed!

**Pull request merged**